powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/service/StepResolutionService.java

```diff
@@ -93,13 +90,17 @@ public StepResolutionService(StepDefinitionRepository stepDefinitionRepository,
      *
      * @param request request to create a new operation
      * @return response with ordered list of next steps
+     * @throws OperationAlreadyExistsException Thrown when operation already exists.
      */
-    public CreateOperationResponse resolveNextStepResponse(CreateOperationRequest request) {
+    public CreateOperationResponse resolveNextStepResponse(CreateOperationRequest request) throws OperationAlreadyExistsException {
         CreateOperationResponse response = new CreateOperationResponse();
         if (request.getOperationId() != null && !request.getOperationId().isEmpty()) {
             // operation ID received from the client, verify that it is available
-            if (operationPersistenceService.getOperation(request.getOperationId()) != null) {
-                throw new IllegalArgumentException("Operation could not be created, operation ID is already used: " + request.getOperationId());
+            try {
+                operationPersistenceService.getOperation(request.getOperationId());
```

Comment:
I don't like this pattern very much - we are throwing exception in case everything is correct and ignore exception in case of unusual state. I suggest implementing `bool operationExists(String operationId)` method on repository level and use it to implement this in a bit cleaner manner...

Reply:
Good idea, I will fix it.